File: Python/problem002.py
#! /usr/bin/env python3
"""
Problem 2 - Project Euler
http://projecteuler.net/index.php?section=problems&id=002
"""


def fib(x):
    a, b = 1, 2
    while a < x:
        yield a
        a, b = b, a + b


if __name__ == "__main__":
    print(sum(x for x in fib(4_000_000) if x % 2 == 0))

File: Nim/problem002.nim
# Problem 2 - Project Euler
# http://projecteuler.net/index.php?section=problems&id=2

proc fibevensum(max: int): int =
    var a = 1
    var b = 2
    while b < max:
        if b mod 2 == 0:
            result += b
        b = a + b
        a = b - a

echo fibevensum(4_000_000)
